pkg/storage/stores/tsdb/single_file_index.go

```diff
@@ -187,20 +223,6 @@ func (i *TSDBIndex) ForSeries(ctx context.Context, shard *index.ShardAnnotation,
 
 }
 
-func (i *TSDBIndex) forPostings(
-	_ context.Context,
-	shard *index.ShardAnnotation,
-	_, _ model.Time,
-	matchers []*labels.Matcher,
-	fn func(index.Postings) error,
-) error {
-	p, err := PostingsForMatchers(i.reader, shard, matchers...)
-	if err != nil {
-		return err
-	}
-	return fn(p)
-}
```

Comment:
I would extract this function into an interface (you already have the `PostingsClient` interface).